elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ToElasticsearch.java

```diff
@@ -255,6 +259,20 @@ private static JsonObject convertMatchAllDocsQuery(MatchAllDocsQuery matchAllDoc
 		return JsonBuilder.object().add( "match_all", new JsonObject() ).build();
 	}
 
+	private static JsonObject convertMatchNoDocsQuery(MatchNoDocsQuery matchNoDocsQuery) {
```

Comment:
You're right. We managed to avoid it until now though, and if we can avoid it until 6.0 we may not even have to do it at all: I'm not sure we'll want to support Lucene => ES conversion once Hibernate Search APIs don't depend on Lucene anymore.